src/main/java/com/epam/idea/core/service/impl/IdeaServiceImpl.java

```diff
@@ -83,21 +93,34 @@ public Idea update(final long ideaId, final Idea source) {
 	@Override
 	@Transactional(readOnly = true)
 	public List<Idea> findIdeasByUserId(final long userId) {
-		return ideaRepository.findByUserId(userId);
+		boolean isAnonymous = SecurityContextHolder.getContext().getAuthentication() instanceof AnonymousAuthenticationToken;
+		List<Idea> ideas = ideaRepository.findByUserId(userId);
+		ideas.forEach(idea -> {
+			Hibernate.initialize(idea.getRelatedTags());
+			if (isAnonymous) {
+				idea.setLiked(isCurrentUserLikedIdea(idea.getId()));
+			}
+		});
+		return ideas;
 	}
 
 	@Override
 	@Transactional(readOnly = true)
 	public List<Idea> findIdeasByTagId(final long tagId) {
+		boolean isAnonymous = SecurityContextHolder.getContext().getAuthentication() instanceof AnonymousAuthenticationToken;
 		final List<Idea> ideas = ideaRepository.findByTagId(tagId);
-		ideas.forEach(idea -> Hibernate.initialize(idea.getRelatedTags()));
+		ideas.forEach(idea -> {
+			Hibernate.initialize(idea.getRelatedTags());
+			if (isAnonymous) {
+				idea.setLiked(isCurrentUserLikedIdea(idea.getId()));
+			}
+		});
 		return ideas;
 	}
```

Comment:
May be, it makes sense to move logic "boolean isAnonymous = Securi... if (isAnonymous) ...idea.setLiked(..." to separate method?